packages/notifications/push/amplify_push_notifications/test/amplify_push_noitfications_impl_test.dart

```diff
@@ -0,0 +1,444 @@
+      );
+      await plugin.configure(
+        authProviderRepo: authProviderRepo,
+        config: config,
+      );
+    });
+    test('identifyUser', () {
+      plugin.identifyUser(
+        userId: 'test-user-1',
+        userProfile: AnalyticsUserProfile(),
+      );
+      verify(
+        mockServiceProviderClient.identifyUser(
+          userId: anyNamed('userId'),
+          userProfile: anyNamed('userProfile'),
+        ),
+      ).called(1);
+    });
+  });
+
+  group('Notification handling APIs', () {
+    setUp(() {
+      testWidgetsFlutterBinding = TestWidgetsFlutterBinding.ensureInitialized();
+      plugin = AmplifyPushNotifications(
+        serviceProviderClient: mockServiceProviderClient,
+        backgroundProcessor: () async => {},
+        dependencyManager: dependencyManager,
+      );
+    });
+    test(
+        'onNotificationReceivedInBackground throws an Exception when the given callback function is not top-level or static',
+        () async {
+      runZoned(
+        () {
+          plugin = AmplifyPushNotifications(
+            serviceProviderClient: mockServiceProviderClient,
+            backgroundProcessor: () async => {},
+          );
+
+          expect(
+            () async => plugin.onNotificationReceivedInBackground(
+              (testGlobalCallbackFunction) {},
+            ),
+            throwsA(const TypeMatcher<PushNotificationException>()),
+          );
+        },
+        zoneValues: {
+          androidCheckOverride: true,
+        },
+      );
+    });
+
+    test(
+        'onNotificationReceivedInBackground should register a top-level or static callback function',
+        () async {
+      await runZoned(
+        () async {
+          SharedPreferences.setMockInitialValues({});
+          final pref = await SharedPreferences.getInstance();
+          plugin = AmplifyPushNotifications(
+            serviceProviderClient: mockServiceProviderClient,
+            backgroundProcessor: () async => {},
+          )..onNotificationReceivedInBackground(
+              testGlobalCallbackFunction,
+            );
+          await Future.delayed(const Duration(seconds: 3), () {});
+
+          expect(pref.containsKey(externalHandleKey), true);
+        },
+        zoneValues: {
+          androidCheckOverride: true,
+        },
+      );
+    });
+
+    test('should invoke the top-level or static external callback function',
+        () async {
+      await runZoned(
+        () async {
+          SharedPreferences.setMockInitialValues({});
+          final pref = await SharedPreferences.getInstance();
+          plugin = AmplifyPushNotifications(
+            serviceProviderClient: mockServiceProviderClient,
+            backgroundProcessor: () async => {},
+            dependencyManager: dependencyManager,
+          )..onNotificationReceivedInBackground(
+              testGlobalCallbackFunction,
+            );
+          await Future.delayed(const Duration(seconds: 3), () {});
```

Comment:
nit: Does it really need 3 secs?

```suggestion
          await Future<void>.delayed(const Duration(seconds: 3));
```

Reply:
No, it's just some arbitrary time I put in to check. Can be lesser.